python/src/rules_engine/parser.py

```diff
@@ -117,15 +175,19 @@ def parse_gas_bill(
     Tries to automatically detect the company that sent the bill.
     Otherwise, requires the company be passed as an argument.
     """
-    data_without_newlines = _newline_line_ending(data)
+    data = _newline_line_ending(data)
+    data = _remove_double_quotes(data)
+    data = _replace_tabs_with_commas(data)
+    data = _remove_non_ascii_or_newline_characters(data)
+
     if company == None:
-        company = _detect_gas_company(data_without_newlines)
+        company = _detect_gas_company(data)
 
     match company:
         case NaturalGasCompany.EVERSOURCE:
-            return _parse_gas_bill_eversource(data_without_newlines)
+            return _parse_gas_bill_eversource(data)
         case NaturalGasCompany.NATIONAL_GRID:
-            return _parse_gas_bill_national_grid(data_without_newlines)
+            return _parse_gas_bill_national_grid(data)
         case _:
             raise ValueError("Wrong CSV format selected: select another format.")
```

Comment:
TODO: Clarify what "wrong csv format" means in this error message. Or make a more appropriate message like, "Something isn't working with this CSV and we don't know what. Here's a link to what your file can look like."